applications/RomApplication/tests/cpp_tests/test_pg_rom_builder_and_solver.cpp

```diff
@@ -0,0 +1,233 @@
+        rResult.reserve(NumNodes);
+
+        for (const auto& r_node : GetGeometry()) {
+            rResult.push_back(r_node.GetDof(TEMPERATURE).EquationId());
+        }
+    }
+
+    void GetDofList(DofsVectorType& rElementalDofList,
+                    const ProcessInfo& rCurrentProcessInfo) const override
+    {
+        rElementalDofList.clear();
+        rElementalDofList.reserve(NumNodes);
+
+        for (const auto& r_node : GetGeometry()) {
+            rElementalDofList.push_back(r_node.pGetDof(TEMPERATURE));
+        }
+    }
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                              VectorType& rRightHandSideVector,
+                              const ProcessInfo& rCurrentProcessInfo) override
+    {
+        if (rLeftHandSideMatrix.size1() != NumNodes) {
+            rLeftHandSideMatrix.resize(NumNodes, NumNodes, false);
+        }
+        if (rRightHandSideVector.size() != NumNodes) {
+            rRightHandSideVector.resize(NumNodes, false);
+        }
+
+        // Laplacian
+        BoundedMatrix<double, NumNodes, NumNodes> A;
+        A(0,0) = 1;
+        A(1,0) = -1;
+        A(0,1) = -1;
+        A(1,1) = 1;
+
+        // Uniform source term
+        BoundedVector<double, NumNodes> b;
+        b(0) = 0.5;
+        b(1) = 0.5;
+
+        // Prev step solution
+        BoundedVector<double, NumNodes> x;
+        x[0] = GetGeometry()[0].GetSolutionStepValue(TEMPERATURE);
+        x[1] = GetGeometry()[1].GetSolutionStepValue(TEMPERATURE);
+
+        noalias(rLeftHandSideMatrix) = A;
+        noalias(rRightHandSideVector) = b - prod(A,x);
+    }
+
+private:
+    static constexpr IndexType NumNodes = 2;
+};
+
+
+ModelPart& FillModel(Model& model)
+{
+    auto & mpart = model.CreateModelPart("main");
+    mpart.CreateNewProperties(0);
+    mpart.SetBufferSize(1);
+    mpart.AddNodalSolutionStepVariable(TEMPERATURE);
+
+    mpart.CreateNewNode(1, 0.0, 0.0, 0.0)->AddDof(TEMPERATURE);
+    mpart.CreateNewNode(2, 1.0, 0.0, 0.0)->AddDof(TEMPERATURE);
+    mpart.CreateNewNode(3, 2.0, 0.0, 0.0)->AddDof(TEMPERATURE);
+
+    mpart.GetNode(1).FastGetSolutionStepValue(TEMPERATURE) = 300;
+    mpart.GetNode(2).FastGetSolutionStepValue(TEMPERATURE) = 300;
+    mpart.GetNode(3).FastGetSolutionStepValue(TEMPERATURE) = 300;
+
+    // Elements
+    using LineType = Line2D2<Node<3>>;
+    auto pProperties = mpart.pGetProperties(0);
+    for(std::size_t i=1; i<=2; ++i)
+    {
+        auto p_geometry = std::make_shared<LineType>(mpart.pGetNode(i), mpart.pGetNode(i+1));
+        auto pelem = Kratos::make_intrusive<DummyLaplacianElement>(i, p_geometry, pProperties);
+        mpart.AddElement(pelem);
+    }
+
+    // Basis
+    const auto phi_1 = [](Node<3> const&){ return 1.0; };
+    const auto phi_2 = [](Node<3> const& r_node){ return r_node.X(); };
+    Matrix phi_3 = ZeroMatrix(1,3);
+    // Setting the 3rd mode to be de residual
+    phi_3(0,0) = 0.5;
+    phi_3(0,1) = 1.0;
+    phi_3(0,2) = 0.5;
+
```

Comment:
When using the QR, it is no longer necessary to be orthonormal. The scope of these tests is to test the Builder and Solver regardless of the physics, though it would be preferable for them to be always consistent. I believe this could be changed in a future PR, so I'll add it to the list.